### PR TITLE
feat(Notifications): add snackbar notifications

### PR DESCRIPTION
**Pull request by lucbpz**
## What does this PR do?

- Changes current notifications and uses Snackbars.
  - This gives us the ability of adding custom links the user can click after an action has finished.
  - Only created volumes and used images are displayed within Snackbars.

## Notes for the reviewer

- Please review each action in case there is some link that is not working. 

## Screenshot

image

### CODE REVIEW

**Review by benja-M-1 — CHANGES_REQUESTED** [review]

File: ui/src/hooks/useExportToImage.ts [hunk 1]
```diff
@@ -1,30 +1,38 @@
...
-        ddClient.desktopUI.toast.success(
-          `Volume ${selectedVolumeName} exported to ${imageName}`
+        sendNotification(
+          `Volume ${selectedVolumeName} exported to ${imageName}`,
+          {
+            name: "See image",
+            onClick: async () => {
+              const [_, tag] = imageName.split(":");
+              const image = (await ddClient.docker.cli.exec('image', ['inspect', imageName])).parseJsonObject();
+              ddClient.desktopUI.navigate.viewImage(image[0].Id, tag || "latest");
```
[the remark this answers is not in the document]
Comment (lucbpz) [reply ui/src/hooks/useExportToImage.ts]: Fixed in 0ec95d7240fac2af05f95d7d7c4e972d213badfe

File: ui/src/components/CloneDialog.tsx [hunk 2]
```diff
@@ -87,7 +87,7 @@ export default function CloneDialog({ ...props }) {
   };
 
   return (
-    <Dialog open={props.open} onClose={props.onClose}>
+    <Dialog fullWidth maxWidth="sm" open={props.open} onClose={props.onClose}>
```
Comment (benja-M-1) [inline ui/src/components/CloneDialog.tsx:90]: I would get rid of the text line 102 as it is redundant with the title

File: ui/src/hooks/useExportToImage.ts [hunk 3]
```diff
@@ -11,6 +11,17 @@ export const useExportToImage = () => {
   const context = useContext(MyContext);
   const selectedVolumeName = context.store.volume?.volumeName;
 
+  const navigateToImage = (imageName: string, imageId: string, tag: string) => {
+    ddClient.desktopUI.navigate
+      .viewImage(imageId, tag || "latest")
+      .catch(() => {
+        sendNotification(`Couldn't navigate to image ${imageName}`, {
+          name: "Try again",
+          onClick: () => navigateToImage(imageName, imageId, tag),
```
Comment (benja-M-1) [inline ui/src/hooks/useExportToImage.ts:20]: The users need to dismiss the notification at some point, maybe they don't want to retry indefinitely :)

**Review by benja-M-1 — CHANGES_REQUESTED** [review]

File: ui/src/hooks/useExportToImage.ts (hunk 3 above)
Comment (lucbpz) [reply ui/src/hooks/useExportToImage.ts]: added in 3974eda3be25ace549f1e4381c9e1c2f3803428e

File: ui/src/components/CloneDialog.tsx (hunk 2 above)
Comment (lucbpz) [reply ui/src/components/CloneDialog.tsx:92]: I am waiting for design input on this dialog, before making any change.

**Comment by felipecruz91** [comment]
@lucbpz Do we plan to use different colours depending on the type of backend response? e.g. green for `success` (2XX) or `error` otherwise.

**Comment by benja-M-1** [comment]
On light mode the text is white on white
Screenshot 2022-08-22 at 11 39 44

**Comment by lucbpz** [comment]
> @lucbpz Do we plan to use different colours for the snackbar depending on the type of backend response? e.g. green for `success` (2XX) or red for `error` otherwise.

@felipecruz91 Added red colors in 5c7dc8e68c6e6ccff5cdc37cee1086a565cad79d to the notifications. Take note the colors are not exactly as they will be in the future (we shouldn't have different color keys for dark/white modes, so I unify them into the same color key for both info and error notifications).

**Review by felipecruz91 — APPROVED** [review]

**Comment by felipecruz91** [comment]
@benja-M-1 I'm merging it as I reckon the redundant title is a minor thing that can be done in a follow-up PR once we hear back from the design team.

**Pull request merged**